pkg/api/api.go

```diff
@@ -492,3 +497,77 @@ func DataExport(c *gin.Context) {
 		badRequest(c, err)
 	}
 }
+
+func DataImport(c *gin.Context) {
+	req := c.Request
+	req.ParseMultipartForm(0)
+	table := req.FormValue("table")
+
+	field_delimiter, err := ParseFieldDelimiter(req.FormValue("field_delimiter"))
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	var file multipart.File
+	file, _, err = req.FormFile("file")
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	defer file.Close()
+	reader := csv.NewReader(file)
+	reader.Comma = field_delimiter
+
+	header, err := reader.Read()
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	data, err := reader.ReadAll()
+	if err != nil {
+		log.Print(err)
+		c.JSON(400, Error{err.Error()})
+		return
+	}
+
+	db := DB(c)
+	createQuery := statements.CreateNewTableQuery(table, header)
+	insertQuery := statements.GenerateBulkInsertQuery(table, header, len(data))
+
+	_, err = db.NewTable(createQuery)
+	if err != nil {
+		log.Print(err)
+		c.JSON(500, Error{err.Error()})
+		return
+	}
+
+	result, err := db.BulkInsert(insertQuery, statements.Flatten(data))
```

Comment:
We should cleanup that newly created table if the insert fails. Alternatively you can perform the import in a single transaction so if something fails the whole import is rolled back and there would not be any unwanted artifacts.

Reply:
> Alternatively you can perform the import in a single transaction
There is a contradiction. If a table is really huge, it is impractical to populate the entire table in a single transaction (huge transactions are not good generally). If we use multiple transactions, then we must issue an additional "truncate table" or "drop table" statement explicitly. But what if the failure was due to lost sql server connection? In this case we can't enforce cleanup, so the state of the table after failure depends on the nature of failure. 

So if I was to decide, I would not do that. If something failed, I would leave cleaning table up to the user. I would just document that and that's it. So, failures get simpler to comprehand: whenever failure happened, user knows that (s)he must clean the table by himself.

Reply:
Im fine with your proposal, that approach would be similar to the sql import (using `psql`) and in case of an error it's up to the user to decide what to do next.